### PR TITLE
Migrate KEB from dep to gomod

### PR DESCRIPTION
**Pull request by wozniakjan**
**Description**
https://github.com/orgs/kyma-project/projects/10#card-60555684

Changes proposed in this pull request:

- remove dep and add go modules

**Related issue(s)**

### CODE REVIEW

File: components/kyma-environment-broker/go.mod [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+	golang.org/x/mod v0.4.2
+	golang.org/x/oauth2 v0.0.0-20210514164344-f6687ab2804c
+	gopkg.in/yaml.v2 v2.4.0
+	k8s.io/api v0.20.5
+	k8s.io/apimachinery v0.20.5
+	k8s.io/client-go v11.0.1-0.20190409021438-1a26190bd76a+incompatible
+	sigs.k8s.io/controller-runtime v0.8.3
+)
+
+replace (
```
Comment (wozniakjan) [inline components/kyma-environment-broker/go.mod:49]: sadly, I wasn't able to get rid of the following `replace` directives. An explanation provided for each and hopefully, gradually we can remove them one by one.

File: components/kyma-environment-broker/Makefile [hunk 2]
```diff
@@ -12,10 +12,36 @@ include $(SCRIPTS_DIR)/generic_make_go.mk
...
+go-mod-check: go-mod-check-local
+go-mod-check-local:
+	@echo make go-mod-check
+	go mod tidy
+	@if [ -z "$$(git status -s go.*)" ]; then \
+		echo -e "${RED}✗ go mod tidy modified go.mod or go.sum files${NC}"; \
+		git status -s git status -s go.*; \
+		exit 1; \
+	fi;
+
```
Comment (wozniakjan) [inline components/kyma-environment-broker/Makefile:44]: this is pretty much copy/paste from control-plane/component/provisioner

**Review by jasiu001 — APPROVED** [review]
Tests and pipelines work, testing provisioning on mps-dev works correctly. LGTM.

**Review by jasiu001 — APPROVED** [review]

**Pull request merged**